server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -379,19 +378,22 @@ void removeTombstoneUnderLock(BytesRef uid) {
     void pruneTombstones(long currentTime, long pruneInterval) {
         for (Map.Entry<BytesRef, DeleteVersionValue> entry : tombstones.entrySet()) {
             BytesRef uid = entry.getKey();
-            try (Releasable ignored = acquireLock(uid)) { // can we do it without this lock on each value? maybe batch to a set and get
-                // the lock once per set?
-                // Must re-get it here, vs using entry.getValue(), in case the uid was indexed/deleted since we pulled the iterator:
-                DeleteVersionValue versionValue = tombstones.get(uid);
-                if (versionValue != null) {
-                    // check if the value is old enough to be removed
-                    final boolean isTooOld = currentTime - versionValue.time > pruneInterval;
-                    if (isTooOld) {
-                        // version value can't be removed it's
-                        // not yet flushed to lucene ie. it's part of this current maps object
-                        final boolean isNotTrackedByCurrentMaps = versionValue.time < maps.getMinDeleteTimestamp();
-                        if (isNotTrackedByCurrentMaps) {
-                            removeTombstoneUnderLock(uid);
+            Releasable lock = keyedLock.tryAcquire(uid);
+            if (lock != null) {
+                try (Releasable ignored = lock) { // can we do it without this lock on each value? maybe batch to a set and get
```

Comment:
Does it make sense to call `tryAcquire` in the try-with-resources block and do the null check in the block? i.e.:

```java
try (Releasable lock = keyedLock.tryAcquire(uid)) {
    if (lock != null) {
        DeleteVersionValue versionValue = tombstones.get(uid);
        // ...
    }
}
```

That implementation would not need a dummy `ignored` variable and instead actually use the returned value (i.e. `lock`).

Reply:
it does I didn't know you can do that with a null value.